lib/collection/array.js

```diff
@@ -240,9 +242,8 @@ export class ArrayColl extends KeyValueCollection {
   }
 
   pop() {
-    let removedItem;
-    if (this._array.length) {
-      removedItem = this._array.pop();
+    let removedItem = this._array.pop();
+    if (removedItem) {
```

Comment:
This reintoduces the bug I was trying to fix.

Reply:
I don't know how that happened. I didn't write this comment recently.